tests/test_queue.py

```diff
@@ -476,3 +484,62 @@ def test_ApiJobQueue_stop_results_in_queue_threads_not_running(mocker):
 
         job_queue.main_thread.quit.assert_called_once_with()
         job_queue.download_file_thread.quit.assert_called_once_with()
+
+
+def test_ApiJobQueue_emits_main_queue_updated_signal_when_message_or_reply_download_added(
+    mocker,
+):
+    mock_client = mocker.MagicMock()
+    mock_session_maker = mocker.MagicMock()
+
+    with threads(2) as [main_thread, file_download_thread]:
+        job_queue = ApiJobQueue(mock_client, mock_session_maker, main_thread, file_download_thread)
+
+    message_download_job = MessageDownloadJob("mock", "mock", "mock")
+    reply_download_job = ReplyDownloadJob("mock", "mock", "mock")
+    main_queue_updated_emissions = QSignalSpy(job_queue.main_queue.queue.queue_updated_signal)
+    job_queue.main_thread.isRunning = mocker.MagicMock(return_value=True)
+    job_queue.download_file_thread.isRunning = mocker.MagicMock(return_value=True)
+
+    job_queue.enqueue(message_download_job)
+    job_queue.enqueue(reply_download_job)
+    main_queue_updated_emissions.wait(MAX_SIGNAL_WAITING_TIME)
```

Comment:
```suggestion
    job_queue.enqueue(message_download_job)
    job_queue.enqueue(reply_download_job)
    main_queue_updated_emissions.wait(MAX_SIGNAL_WAITING_TIME)
    main_queue_updated_emissions.wait(MAX_SIGNAL_WAITING_TIME)
```

If enqueuing two jobs, I think it's prudent waiting for two signals. In practice the signals are processed extremely quickly in a small test like this, and granted the double waiting is not the most elegant... but there is a race condition, and waiting double doesn't actually make the test slower.

Truth be told, you might not need to wait _at all_ for the test to pass, but I'd rather treat the signals as if it weren't _almost-synchrounous_ in practice, so that their caveats get printed in our minds when reading the code.